### PR TITLE
Update logistic_regression_categorical_encoding.ipynb

### PR DESCRIPTION
**Pull request by alvinwalker314**

### CODE REVIEW

**Review by hughjafro — COMMENTED** [review]
Module 1:

- Start a clean notebook.
  - Good use of comments
- Do train/validate/test split with the Tanzania Waterpumps data.
  - Stretch: Explore different train_size and test_size
- Begin to explore and clean the data.
  - Import has the correct shape
  - Good methodical approach to analyze each feature
  - I would be interested to know your thought process in creating the function return_mean_if_zero
  - Good start to analyze for cardinality

- Select different numeric and categorical features.
  - Your cat features selections are a good choice
- Do one-hot encoding. (Remember it may not work with high cardinality categoricals.)
- Scale features.
- Use scikit-learn for logistic regression.
  - Check
  - Does using `solver='liblinear' in your LR model make a difference in score?
- Get your validation accuracy score.
  - nice score
- Get and plot your coefficients.
  - How you might address and impute missing/zero values have an effect on model scores, try different ways
- Submit your predictions to our Kaggle competition.
- Commit your notebook to your fork of the GitHub repo.

**Review by hughjafro — COMMENTED** [review]
### Module 02 - Decision Trees
**Overall:** Well done getting in the Top 10 📈
**Feedback:** Push yourself to break 80%. Try using different models like GradientBoosting and RandomForest

#### More Exploring/Cleaning data
Well done wrangling your data!

#### Implement a Decision Tree Classifier
I did not see a DT model used. Make sure that you also focus on feature selection/engineering.

Submit New Predictions to Kaggle
👍

**Comment by hughjafro** [comment]
### Module 03 Random Forest:

**Overall:** You are doing an incredible job. Getting to 81% is tough!!!
**Feedback:** Try to push yourself to get in the top 3. Nice commenting.

_Do more exploratory data analysis, data cleaning, feature engineering, and feature selection._
- Experiment with LabelEncoder to encode labels on ytrain and Onehot for categoricals.
- Experiment with different feature selections

_Try a Random Forest Classifier._
- Check

_Try Ordinal Encoding._
- Check

What other models have you tried? GBM? XGboost?

**Submit new predictions.**

**Comment by hughjafro** [comment]
### Module 04 Confusion Matrix:

**Overall:** Good job this week!!!
**Feedback:** Create a kernel with more text explanation

_Get and interpret the confusion matrix for classification models_
- At least try to implement a confusion matrix 😕 
- What does the confusion matrix tell you?
- What is the significance of the precision, recall, f-1 scores?

_Kaggle Challenge_
- What is the most important thing that you learned in this Kaggle challenge? Write a blog about it ✍️ .
- Did you have problems with overfitting or underfitting? If so, how was it handled?
- Ask yourself, what you would do differently on your next Kaggle challenge? In approach and execution.
- Were there any models that you did not try to implement that you wish you did?

**You are doing awesome**
